ubtue_nirc.js

```diff
@@ -102,10 +102,10 @@ function invokeEmbeddedMetadataTranslator(doc, url) {
 				item.creators.push({"firstName": firstname, "lastName": lastname, "creatorType": "author"});
 			}
 		}
-		if (ZU.xpathText(doc, '//main//h1//a')) {
-			let issueinfo = ZU.xpathText(doc, '//main//h1//a');
-			item.volume = issueinfo.match(/Volume (\d+)/)[1];
-			item.issue = issueinfo.match(/Issue (\d+)/)[1];
+		if (ZU.xpathText(doc, '//main//h1//a[contains(text(), "Volume") or contains(text(), "Issue")]')) {
+			let issueinfo = ZU.xpathText(doc, '//main//h1//a[contains(text(), "Volume") or contains(text(), "Issue")]');
+			item.volume = issueinfo.match(/Volume (\d+)/) ? issueinfo.match(/Volume (\d+)/)[1] : null;
+			item.issue = issueinfo.match(/Issue (\d+)/) ? issueinfo.match(/Issue (\d+)/)[1] : null;
 		}
```

Comment:
```js
		if (ZU.xpathText(doc, '//main//h1//a')) {
			let issueinfo = ZU.xpathText(doc, '//main//h1//a');
			let volumeMatch = issueinfo.match(/Volume (\d+)/);
			item.volume = volumeMatch ?  volumeMatch[1] : null;
			let issueMatch = issueinfo.match(/Issue (\d+)/);
			item.issue = issueMatch ? issueinfo.match(/Issue (\d+)/)[1] : null;
```
Makes it easier to integrate potential additional cases.
To be very expliciz one might use something like `(issueMatch && issueMatch.length > 1) ? 			               issueinfo.match(/Issue (\d+)/)[1] : null;` but due to the brackets the size of the result array must be implictly > 1 if a match occurs.

Reply:
```
if (ZU.xpathText(doc, '//main//h1//a')) {
            let issueinfo = ZU.xpathText(doc, '//main//h1//a');
            let volumeMatch = issueinfo.match(/Volume (\d+)/);
            item.volume = volumeMatch ?  volumeMatch[1] : null;
            let issueMatch = issueinfo.match(/Issue (\d+)/);
            item.issue = issueMatch ? issueMatch[1] : null;
 }
```

Is the second Issue Regex really necessary?
Are you referring to the brackets in my original if statement?

Reply:
> Is the second Issue Regex really necessary?

No, was meant to be `issueinfo[1]`
> Are you referring to the brackets in my original if statement?

No to the grouping brackets in the regex - in general it cannot be assumed the an array has more than one element if it is not empty, but in this specific regex case this hold true.

Reply:
```suggestion
let issueinfo = ZU.xpathText(doc, '//main//h1//a');
if (issueinfo) {
    let volumeMatch = issueinfo.match(/Volume (\d+)/);
    item.volume = volumeMatch ?  volumeMatch[1] : null;
    let issueMatch = issueinfo.match(/Issue (\d+)/);
    item.issue = issueMatch ? issueMatch[1] : null;
}
```

I was originally working with the given conventions in this translator code, but would prefer this solution - do you have any objections?

Reply:
Ok.